### PR TITLE
feat: add TextInput component to generalize re-used text input style …

### PR DESCRIPTION
**Pull request by codyconway77**
…and labeling and refactor signup and login to use TextInput

### CODE REVIEW

File: src/app/components/TextInput/TextInput.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+            name={name}
+            type={type}
+            value={value}
+            onChange={onChange}
+            required={required}
+            autoComplete={autoComplete}
+            className={`mt-1 block w-full px-3 py-2 border border-gray-300 rounded-md shadow-sm placeholder-gray-400 focus:outline-none focus:ring-indigo-500 focus:border-indigo-500 sm:text-sm ${className}`}
+            placeholder={placeholder}
+        />
+    </>
```
Comment (codyconway77) [inline src/app/components/TextInput/TextInput.tsx:44]: Formatting is messed up

File: src/app/components/SignupForm/SignupForm.tsx [hunk 2]
```diff
@@ -32,46 +33,47 @@ const SignupForm = () => {
...
           type="password"
-          required
+          autoComplete="current-password"
+          required={true}
           value={password}
           onChange={(e) => setPassword(e.target.value)}
-          className="mt-1 block w-full rounded-md border-gray-300 shadow-sm focus:border-indigo-500 focus:ring-indigo-500"
         />
       </div>
       <div>
```
Comment (codyconway77) [inline src/app/components/SignupForm/SignupForm.tsx:59]: These outer `<div>` can likely be moved to the `TextInput` component

**Comment by codyconway77** [comment]
Potential Enhancements:
- `password` type inputs having an "eyeball" icon for show/hide password
- `disabled` styling
- add `invalid` and its styles

**Review by nicktran6997 — APPROVED** [review]
We should think about how we want to structure those tailwind css's into a global reusable class but i'll put that in the backlog

**Pull request merged**